tests/sentry/api/endpoints/test_auth_index.py

```diff
@@ -134,39 +134,101 @@ def test_superuser_sso(self):
         with mock.patch.object(Superuser, "org_id", self.organization.id), override_settings(
             SUPERUSER_ORG_ID=self.organization.id
         ):
-            self.login_as(user, organization_id=self.organization.id)
+            with self.settings(SENTRY_SELF_HOSTED=False):
+                self.login_as(user, organization_id=self.organization.id)
+                response = self.client.put(
+                    self.path,
+                    data={
+                        "isSuperuserModal": True,
+                        "superuserAccessCategory": "for_unit_test",
+                        "superuserReason": "for testing",
+                    },
+                )
+                assert response.status_code == 200
+                assert COOKIE_NAME in response.cookies
+
+    def test_superuser_sso_user_has_password_saas_product(self):
+        from sentry.auth.superuser import COOKIE_NAME, Superuser
+
+        org_provider = AuthProvider.objects.create(organization=self.organization, provider="dummy")
+
+        user = self.create_user("foo@example.com", is_superuser=True)
+
+        AuthIdentity.objects.create(user=user, auth_provider=org_provider)
+
+        with mock.patch.object(Superuser, "org_id", self.organization.id), override_settings(
+            SUPERUSER_ORG_ID=self.organization.id
+        ):
+            with self.settings(SENTRY_SELF_HOSTED=False):
+                self.login_as(user, organization_id=self.organization.id)
+                response = self.client.put(
+                    self.path,
+                    data={
+                        "isSuperuserModal": True,
+                        "superuserAccessCategory": "for_unit_test",
+                        "superuserReason": "for testing",
+                    },
+                )
+                assert response.status_code == 200
+                assert COOKIE_NAME in response.cookies
+
+    def test_superuser_no_sso_saas_product(self):
+        from sentry.auth.superuser import Superuser
+
+        AuthProvider.objects.create(organization=self.organization, provider="dummy")
+
+        user = self.create_user("foo@example.com", is_superuser=True)
+
+        with mock.patch.object(Superuser, "org_id", self.organization.id), override_settings(
+            SUPERUSER_ORG_ID=self.organization.id
+        ):
+            self.login_as(user)
             response = self.client.put(
                 self.path,
                 data={
+                    "password": "admin",
                     "isSuperuserModal": True,
                     "superuserAccessCategory": "for_unit_test",
                     "superuserReason": "for testing",
                 },
             )
-            assert response.status_code == 200
-            assert COOKIE_NAME in response.cookies
+            assert response.status_code == 401
 
-    def test_superuser_no_sso(self):
+    def test_superuser_no_sso_self_hosted(self):
```

Comment:
Same as above, might be better to explicitly mention that you have password here.